pbxproj/pbxextensions/ProjectFiles.py

```diff
@@ -423,6 +423,89 @@ def add_folder(self, path, parent=None, excludes=None, recursive=True, create_gr
 
         return results
 
+    def add_package(self, repositoryURL, product_name, target_name, package_requirement={}):
+        package_ref, is_created = self.get_or_create_package_reference(repositoryURL, (repositoryURL, package_requirement,))
+
+        if is_created:
+            for project_object in self.objects.get_objects_in_section('PBXProject'):
+                if 'packageReferences' not in project_object:
+                    project_object['packageReferences'] = PBXList()
+
+                if package_ref.get_id() not in project_object['packageReferences']:
+                    project_object.packageReferences.append(package_ref.get_id())
+
+        if not isinstance(product_name, list):
+            product_name = [product_name]
+
+        for name in product_name:
+            package_dep, is_created = self.get_or_create_package_dependency(name, (package_ref, name,))
+
+            if is_created:
+                # add build files (PBXBuildFile section and PBXFrameworksBuildPhase section)
+                self._create_build_products(package_dep, target_name)
+
+                # add packageProductDependencies to PBXNativeTarget section
+                target = self.get_target_by_name(name=target_name)
+                if 'packageProductDependencies' not in target:
+                    target['packageProductDependencies'] = PBXList()
+
+                if package_dep.get_id() not in target['packageProductDependencies']:
+                    target.packageProductDependencies.append(package_dep.get_id())
```

Comment:
Similarly, this seems to be part of the `get_or_create_package_dependency` logic. I would rather move them inside that function.

Reply:
Agreed